src/electionguard_cli/e2e_steps/election_record_step.py

```diff
@@ -0,0 +1,52 @@
+from shutil import make_archive
+from os.path import splitext
+import click
+import tempfile
+from electionguard_cli.cli_models import BuildElectionResults, E2eSubmitResults
+from electionguard_cli.cli_models.e2e_decrypt_results import E2eDecryptResults
+from electionguard_cli.cli_models.e2e_inputs import E2eInputs
+from electionguard_cli.e2e_steps.e2e_step_base import E2eStepBase
+from electionguard.constants import get_constants
+
+from electionguard_tools.helpers.export import export_record
+
+
+class ElectionRecordStep(E2eStepBase):
+    """Responsible for publishing an election record after an election has completed."""
+
+    REMOVE_RAW_OUTPUT = True
+    REMOVE_ZIP_OUTPUT = True
+    COMPRESSION_FORMAT = "zip"
+
+    def run(
+        self,
+        election_inputs: E2eInputs,
+        build_election_results: BuildElectionResults,
+        submit_results: E2eSubmitResults,
+        decrypt_results: E2eDecryptResults,
+    ) -> None:
+
+        self.print_header("Election Record")
+        guardian_records = [
+            guardian.publish() for guardian in election_inputs.guardians
+        ]
+        constants = get_constants()
+        with tempfile.TemporaryDirectory() as temp_dir:
+            export_record(
```

Comment:
This file is the only real code in this PR, everything else is just moving stuff around in order to get access to all the bits of data that this step needs.